pkg/arvo/ted/eth-watcher.hoon

```diff
@@ -80,26 +82,94 @@
   |=  [pup=watchpup =latest=number:block]
   =/  m  (strand:strandio ,watchpup)
   ^-  form:m
-  =/  zoom-margin=number:block  30
+  ~&  >  [%zoom-start number.pup latest-number]
+  =/  zoom-margin=number:block  0  :: TODO: 30!
   =/  zoom-step=number:block  100.000
   ?:  (lth latest-number (add number.pup zoom-margin))
     (pure:m pup)
-  =/  up-to-number=number:block  (sub latest-number zoom-margin)
+  =/  up-to-number=number:block
+    (min (add 10.000.000 number.pup) (sub latest-number zoom-margin))
   |-
+  ~&  >  [%zooming number.pup up-to-number]
   =*  loop  $
   ?:  (gth number.pup up-to-number)
     (pure:m pup(blocks ~))
   =/  to-number=number:block
-    (min up-to-number (add number.pup zoom-step))
+    =;  step
+      (min up-to-number (add number.pup step))
+    ::  Between "launch" (6.784.800) and "public" (7.033.765) blocks,
+    ::  there are a lot events belonging to all the pre-ethereum ships
+    ::  being established on-chain. By reducing the step, we avoid crashing.
+    ::
+    ?.  =(contracts:azimuth mainnet-contracts:azimuth)
```

Comment:
@philipcmonk I had to comment this line when getting all blocks on mainnet so it wouldn't use 100.000 blocks all the time (it came from this commit: https://github.com/urbit/urbit//commit/781cdf39606dcad0dcdbd4ae71764ec0aca13aea)

Reply:
I don't understand, isn't the point of this line to make sure that on mainnet it uses smaller block batches?  On mainnet shouldn't this condition evaluate to true?

Reply:
I think i found the issue, we need to change this https://github.com/urbit/urbit//commit/781cdf39606dcad0dcdbd4ae71764ec0aca13aea#diff-f78b5c5642b98a402f139a465bdfcb91309f95ce3c5574b906d12258335ad692R81 to `mainnet-contracts`